File: services/go-worker/plugins/hello_plugin.go
package plugins

import (
	"fmt"
	"time"
)

// HelloPlugin implements a simple hello world capability
type HelloPlugin struct {
	BasePlugin
}

// NewHelloGoPlugin creates a new HelloPlugin instance
func NewHelloGoPlugin() *HelloPlugin {
	return &HelloPlugin{}
}

func (p *HelloPlugin) GetName() string {
	return "hello_go"
}

func (p *HelloPlugin) GetDescription() string {
	return "Returns a hello message from Go worker"
}

func (p *HelloPlugin) Execute(params map[string]interface{}, context *ExecutionContext) (interface{}, error) {
	name := "World"
	if n, ok := params["name"].(string); ok {
		name = n
	}

	return map[string]interface{}{
		"message":    fmt.Sprintf("Hello %s from Go! 🔵", name),
		"worker_id":  context.WorkerID,
		"timestamp":  time.Now().Format(time.RFC3339),
		"go_version": "go1.18",
		"status":     "success",
	}, nil
}

File: services/python-worker/plugins/hello_plugin.py
"""
Hello Plugin - Simple greeting capability
"""

from datetime import datetime
from plugins.base_plugin import BasePlugin


class HelloPlugin(BasePlugin):
    """Simple hello world capability"""
    
    @property
    def name(self) -> str:
        return "hello"
    
    @property
    def description(self) -> str:
        return "Returns a hello message"
    
    @property
    def output_schema(self) -> str:
        return '{"type":"object","properties":{"message":{"type":"string"}}}'
    
    def execute(self, params: dict, worker_sdk=None) -> dict:
        return {
            "message": "Hello World from Python Worker! 🐍",
            "timestamp": datetime.now().isoformat(),
            "status": "success"
        }
